Code/GraphMol/FileParsers/ProximityBonds.cpp

```diff
@@ -47,6 +47,41 @@ static bool IsBonded(ProximityEntry *p, ProximityEntry *q, unsigned int flags) {
   return dist2 <= radius * radius;
 }
 
+bool SamePDBResidue(AtomPDBResidueInfo *p, AtomPDBResidueInfo *q) {
+    return p->getResidueNumber() == q->getResidueNumber() &&
+           p->getResidueName() == q->getResidueName() &&
+           p->getChainId() == q->getChainId() &&
+           p->getInsertionCode() == q->getInsertionCode();
+}
+
+
+static bool IsBlacklistedAtom(Atom *atom){
+    // blacklist metals, noble gasses and halogens
+    unsigned int elem = atom->getAtomicNum();
+    // make an inverse query (non-metals and metaloids)
+    if((5 <= elem && elem <= 8) ||
+       (14 <= elem && elem <= 16) ||
+       (32 <= elem && elem <= 34) ||
+       (51 <= elem && elem <= 52))
+        return false;
+    else
+        return true;
+}
+
+bool IsBlacklistedPair(Atom *beg_atom, Atom *end_atom) {

```

Comment:
Since this can be called from other places, it should start with a `PRECONDITION` for both `beg_atom` and `end_atom`.